### PR TITLE
Copy Makefile and .sass files from kubernetes/website and update them to match our current CSS

### LINKED ISSUE #3: Copy .sass files from kubernetes/website and use overrides.sass
When copying the website across, I copied the .css files instead of the source .sass files.

We should probably use the .sass files and build the resulting .css ourselves. This will allow us to provide an overrides file to make rebasing easier, if we ever do.

### PR DESCRIPTION
**Pull request by xmudrii**
This PR copies the Makefile and .sass files from the kubernetes/website repo.

The Makefile ensures we can easier regenerate the CSS files from .sass files, as well as easier build the website.

The CSS/sass regeneration depends on the [node-sass](https://github.com/sass/node-sass) package, which can be obtained from npm, such as `npm install -g node-sass`. I've added a note about this to Makefile and Makefile help.

The CSS files are regenerated using the `make sass` command.

The .sass files are updated to match our current CSS. Maybe there's something left unnoticed, but I'll try to fix that as well.

Closes #3
/area hugo
/cc @nikhita @LiliC @munnerz

### CODE REVIEW

File: static/css/styles.css [hunk 1]
```diff
@@ -426,23 +426,18 @@ dd { margin-bottom: 16px; }
...
-/* A little extra margin and visual separation for consecutive top-level lists */
-#docsContent>ul, #docsContent>ol { margin-bottom: 40px !important; border-bottom: 1px solid #eee !important; }
+#docsContent ul, #docsContent ol { margin: 20px 0; padding-left: 30px; font-weight: 300; }
 
 #docsContent ul ul, #docsContent ol ol, #docsContent ul ol, #docsContent ol ul { margin: 0.75em 0; }
 
 #docsContent li { margin-bottom: 0.75em; font-size: 16px; line-height: 1.75em; }
 
-/* No bullets for checklists */
-#docsContent ul.task-list>li { list-style-type: none !important; }
```
Comment (xmudrii) [inline static/css/styles.css:444]: This rule is now on line 433.

File: static/css/styles.css [hunk 2]
```diff
@@ -426,23 +426,18 @@ dd { margin-bottom: 16px; }
...
 #docsContent .includecode pre { margin: 0 !important; }
 
-/* These are being clobbered by line 37 and it is affecting <ul> nested inside <ol> */
-#docsContent ul>li { list-style: disc !important; }
+#docsContent ul > li { list-style: disc !important; }
 
-#docsContent ol>li { list-style: decimal !important; }
+#docsContent ol > li { list-style: decimal !important; }
 
-#docsContent ul, #docsContent ol { margin: 20px 0; padding-left: 30px; font-weight: 300; }
```
Comment (xmudrii) [inline static/css/styles.css:434]: The rule is still here, but it's on line 435 now.

File: static/css/styles.css [hunk 3]
```diff
@@ -426,22 +426,19 @@ dd { margin-bottom: 16px; }
...
+#docsContent ul.task-list > li { list-style-type: none !important; }
 
-/* A little extra margin and visual separation for consecutive top-level lists */
-#docsContent>ul, #docsContent>ol { margin-bottom: 40px !important; border-bottom: 1px solid #eee !important; }
```
Comment (xmudrii) [inline static/css/styles.css]: This rule is now on the line 441. The line after was there before, but for some reason GitHub is showing it as a new one.

**Comment by nikhita** [comment]
@xmudrii thank you, thank you, thank you! :)

I'll try to take a look at this later today but  fair warning: my CSS skills are equivalent to

The Essential art of Changing Stuff and Seeing What Happens
https://twitter.com/thepracticaldev/status/720257210161311744

so I'd appreciate if anyone else took a deeper look at it. :smile:

**Comment by xmudrii** [comment]
> I'll try to take a look at this later today but fair warning: my CSS skills are equivalent to
>
> The Essential art of Changing Stuff and Seeing What Happens
> https://twitter.com/thepracticaldev/status/720257210161311744

Same, but the website looks same as before, so I hope I've got it right. :smile:

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,35 @@
+DOCKER       = docker
+HUGO_VERSION = 0.40.3
+DOCKER_IMAGE = client-go-docs
+DOCKER_RUN   = $(DOCKER) run --rm --interactive --tty --volume $(PWD):/src
+
+.PHONY: all build sass build-preview help serve
+
+help: ## Show this help.
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {sub("\\\\n",sprintf("\n%22c"," "), $$2);printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
```
Comment (munnerz) [inline Makefile:9]: This is some serious make-magic... I am not going to pretend I know what this does, but I guess it extracts out the comments after each Make target to print it out nicely? Clever 😄

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+all: build ## Build site with production settings and put deliverables in _site.
+
+# Buidling sass from sources requires the node-sass package, which can be found on GitHub and npm.
+# https://github.com/sass/node-sass
+sass: # Rebuild the SASS source into CSS. Requires node-sass.
+	node-sass --output-style compact ./src/sass/styles.sass ./static/css/styles.css
+	node-sass --output-style compact ./src/sass/case_study_styles.sass ./static/css/case_study_styles.css
```
Comment (munnerz) [inline Makefile:17]: 👍 we can look at creating a 'build' docker image later, as well as setting up a verify stage as part of CI to ensure the generated CSS is up-to-date.

**Comment by munnerz** [comment]
This seems sensible to me, and solves #3 😄

I did not spend much time adjusting the CSS when first porting things across, so I think we should accept this (because progress!) and push ahead! 💯 

(let's also test tide merging for the first time...)
/lgtm
/approve

File: Makefile (hunk 5 above)
Comment (xmudrii) [reply Makefile:17]: Those are great ideas. :+1: 

I can take a look into that once I have some time. That should be interesting. :smile: 

I'll create an issue as a reminder.

**Comment by xmudrii** [comment]
Sounds good! I'll try to poke around the website bit more to fix some issues, when I have some more free time. I'm still a CSS beginner though, but I think I can help with some minor issues. Also, that's a good practice. :smile: 

I created #8 for a comment you left, that's a great idea! I'll try to fix that one as well. :)

**Pull request merged**